db-client-java/src/main/java/com/eventstore/dbclient/ThrowingFunction.java

```diff
@@ -0,0 +1,7 @@
+package com.eventstore.dbclient;
+
+@FunctionalInterface
+interface ThrowingFunction<TInput, TResult, TException extends Throwable> {
```

Comment:
I fail to see the purpose of that interface, along with `ThrowingBiFunction`. Is this mostly for convenience?

Reply:
This is to allow lambda syntax where the expression throws a checked exception.  The standard functional interfaces don't allow for this in Java.  So for instance in GetProjectionResult, I've been able to use an instance of ThrowingBiFunction to hold the deserialization strategy which can throw a JsonProcessingException.